### PR TITLE
changes done

### PR DESCRIPTION
**Pull request by singhthedev**
added image field in video section:- 
model, controller and routes code 
also change same thing in server room controler update api code for image
changed helper code for email and password validation
sort the getAllVideo api for latest entery first in response

### CODE REVIEW

File: src/routes/protectedRoutes.ts [hunk 1]
```diff
@@ -15,9 +15,24 @@ import {
   deleteVideoById,
 } from "../controllers/adminController";
 import { verifyToken } from "../middlewares/authMiddleware";
+import multer from 'multer';
+import bodyParser from "body-parser";
 
 const route = express.Router();
 
+
+route.use(bodyParser.json());
```
Comment (Ashok-tgc) [inline src/routes/protectedRoutes.ts:24]: why body parser separately?

Comment (singhthedev) [reply src/routes/protectedRoutes.ts:24]: removed

File: src/controllers/adminController.ts [hunk 2]
```diff
@@ -285,6 +296,14 @@ export const video = async (req: Request, res: Response) => {
...
+    let secure_url: string | null = null;
+    if (file) {
+      const uploadResponse: UploadApiResponse = await cloudinary.uploader.upload(
+        file.path
+      );
+      secure_url = uploadResponse.secure_url;
+    }
+
     const checkRoomUuid = await RoomId.findOne({ roomUuid: roomId })
     if (checkRoomUuid) {
```
[the remark this answers is not in the document]
Comment (singhthedev) [reply src/controllers/adminController.ts:308]: done

File: src/controllers/adminController.ts [hunk 3]
```diff
@@ -285,6 +296,14 @@ export const video = async (req: Request, res: Response) => {
       });
     }
 
+    let secure_url: string | null = null;
+    if (file) {
+      const uploadResponse: UploadApiResponse = await cloudinary.uploader.upload(
```
[the remark this answers is not in the document]
Comment (singhthedev) [reply src/controllers/adminController.ts:301]: not uploading in a file it just a variable name it's uploading in cloudnary and then we are getting the file path of uploaded document

File: src/controllers/adminController.ts [hunk 4]
```diff
@@ -388,16 +409,32 @@ export const getVideoById = async (req: Request, res: Response) => {
...
     const { id } = req.params;
-    if (!validId.test(id)) {
-      return res.status(404).json({ error: "Invalid ID" });
-    }
     const updatedVideoData = req.body;
+    const file = req.file;
+
+    let secure_url: string | null = null;
+    if (file) {
+      const uploadResponse: UploadApiResponse = await cloudinary.uploader.upload(
```
[the remark this answers is not in the document]
Comment (singhthedev) [reply src/controllers/adminController.ts:417]: not uploading in a file it just a variable name it's uploading in cloudnary and then we are getting the file path of uploaded document

**Pull request merged**